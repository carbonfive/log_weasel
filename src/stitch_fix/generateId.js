import { ulid } from 'ulid';
import { constantCase } from 'constant-case';

const generateId = (key) => {
  if (typeof key === 'undefined') {
    throw new TypeError('LogWeasel generateID requires a key argument');
  }

  return `${ulid()}-${constantCase(key)}-JS`;
};

export default generateId;
