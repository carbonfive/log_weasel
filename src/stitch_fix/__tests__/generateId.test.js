import generateId from '../generateId';

const appName = 'weasel-gang-ui';

describe('generateId', () => {
  it('throws an error if key is not passed', () => {
    expect(() => generateId()).toThrowErrorMatchingSnapshot();
  });

  it('returns a Log Weasel ID with a constant case app name', () => {
    expect(generateId(appName)).toMatch(/^[0-9A-Z]{26}-WEASEL_GANG_UI-JS$/);
  });

  it('returns a different Log Weasel ID each time it is called', () => {
    const firstId = generateId(appName);
    const secondId = generateId(appName);

    expect(firstId).not.toEqual(secondId);
  });
});
