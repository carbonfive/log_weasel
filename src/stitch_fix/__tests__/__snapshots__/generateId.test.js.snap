// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateId throws an error if key is not passed 1`] = `"LogWeasel generateID requires a key argument"`;
